archive/archive.go

```diff
@@ -0,0 +1,146 @@
+package archive
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+)
+
+type fileCreator interface {
+	create(filename string) (*os.File, error)
+}
+type fileOpener interface {
+	open(filename string) (*os.File, error)
+}
+type fileCopier interface {
+	copy(dst io.Writer, src io.Reader) (written int64, err error)
+}
+
+type zipWriter interface {
+	Create(name string) (io.Writer, error)
+	Close() error
+}
+
+type defaultFileHandler struct{}
+
+func (d *defaultFileHandler) open(filename string) (*os.File, error) {
+	return os.Open(filename)
+}
+
+func (d *defaultFileHandler) create(filename string) (*os.File, error) {
+	return os.Create(filename)
+}
+
+func (d *defaultFileHandler) copy(dst io.Writer, src io.Reader) (written int64, err error) {
+	return io.Copy(dst, src)
+}
+
+type ArchiveHandler interface {
+	Open(path string) (io.Writer, error)
+	AppendFileToArchive(path string, writer zip.Writer) error
+	Close() error
+}
+
+// SupportArchiveHandler
+// The normal procedure should look like this.
+// 		1. CreateZipArchiveFile
+// 		2. InitialiseZipWriter
+// 		3. AppendFileToArchive (n times)
+// 		4. Close
+type SupportArchiveHandler struct {
+	writer      zipWriter
+	fileCreator fileCreator
+	fileOpener  fileOpener
+	fileCopier  fileCopier
+}
+
+func New() *SupportArchiveHandler {
+	return &SupportArchiveHandler{
+		fileCreator: &defaultFileHandler{},
+		fileOpener:  &defaultFileHandler{},
+		fileCopier:  &defaultFileHandler{},
+	}
+}
+
+// CreateZipArchiveFile creates the file that will be the zip archive.
+// The zipFilePath expects a complete path with the correct file extension (.zip).
+// If you not intend to create an io.Writer beforehand this method can be the input of InitialiseZipWriter.
+func (ar *SupportArchiveHandler) CreateZipArchiveFile(zipFilePath string) (io.Writer, error) {
+	zippedArchiveFile, err := ar.fileCreator.create(zipFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create archive file: %w", err)
+	}
+	return zippedArchiveFile, nil
+}
+
+// InitialiseZipWriter takes an existing io.Writer and initializes a zip.Writer based on it.
```

Comment:
Wouldnt `initialize` instead of `initialise` the correct spelling? Actually both is correct, but i usually read `initialize` - Whats your opinion?